receiver/otlpreceiver/internal/errors/errors.go

```diff
@@ -23,3 +25,22 @@ func GetStatusFromError(err error) error {
 	}
 	return s.Err()
 }
+
+func GetHTTPStatusCodeFromStatus(err error) int {
+	s, ok := status.FromError(err)
+	if !ok {
+		return http.StatusInternalServerError
+	}
+	// See https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#failures
+	// to see if a code is retryable.
+	// See https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#failures-1
+	// to see a list of retryable http status codes.
+	switch s.Code() {
+	// Retryable
+	case codes.Canceled, codes.DeadlineExceeded, codes.Aborted, codes.OutOfRange, codes.Unavailable, codes.DataLoss:
+		return http.StatusServiceUnavailable
+	// Not Retryable
+	default:
+		return http.StatusInternalServerError
```

Comment:
Permanent errors will be [converted to `status.InvalidArgument`](https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/internal/errors/errors.go#L20), which is equivalent to an HTTP 400. We should change one of the two codes so they're semantically equivalent between gRPC and HTTP.

Reply:
I thought about that too, but since [the spec says](https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#failures-1):

```
This specification does not use Status.code field and the server MAY omit Status.code field. The clients are not expected to alter their behavior based on Status.code field but MAY record it for troubleshooting purposes.
```

I opted for now for 500 since it is what we were returning before.  If we want to change something we should change `GetStatusFromError` to return `codes. INTERNAL` or `codes.UNKNOWN`, but we aren't required to by the spec.

Reply:
With https://github.com/open-telemetry/opentelemetry-collector/pull/9415 merged the status codes are now aligned.